src/components/Images.vue

```diff
@@ -186,72 +171,56 @@ export default {
 			this.$el.onmousemove = null
 		},
 		dragHandler(event) {
-			event.preventDefault()
-			const { pageX, pageY } = event
+			const { clientX, clientY } = event
 
-			if (this.dragging && this.zoomRatio > 1 && pageX > 0 && pageY > 0) {
-				const moveX = this.shiftX + (pageX - this.dragX)
-				const moveY = this.shiftY + (pageY - this.dragY)
-				const growX = this.zoomWidth - this.width
-				const growY = this.zoomHeight - this.height
+			if (this.dragging && this.zoomRatio > 1 && clientX > 0 && clientY > 0) {
+				this.posX += (clientX - this.dragX)
+				this.posY += (clientY - this.dragY)
 
-				this.shiftX = Math.min(Math.max(moveX, -growX / 2), growX / 2)
-				this.shiftY = Math.min(Math.max(moveY, -growY / 2), growX / 2)
-				this.dragX = pageX
-				this.dragY = pageY
+				this.dragX = clientX
+				this.dragY = clientY
 			}
 		},
+
+		/**
+		 * Double click handler
+		 */
 		onDblclick() {
 			if (this.zoomRatio > 1) {
 				this.resetZoom()
 			} else {
+				this.animateTransition = true
+				// 0.15 is half of the delta in zoom,
+				// so that multiplied by width/height centers the image
+				this.posX = -(this.$el.width * 0.15)
+				this.posY = -(this.$el.height * 0.15)
 				this.zoomRatio = 1.3
+				// Animation is 100ms so give it double the time to finish
+				setTimeout(() => { this.animateTransition = false }, 200)
 			}
 		},
 	},
 }
 </script>
 
 <style scoped lang="scss">
-$checkered-size: 8px;
-$checkered-color: #efefef;
-
 img {
-	max-width: 100%;
-	max-height: 100%;
 	align-self: center;
 	justify-self: center;
-	// black while loading
-	background-color: #000;
-	// animate zooming/resize
-	transition: height 100ms ease,
-		width 100ms ease,
-		margin-top 100ms ease,
-		margin-left 100ms ease;
-	// show checkered bg on hover if not currently zooming (but ok if zoomed)
-	&:hover {
-		background-image: linear-gradient(45deg, #{$checkered-color} 25%, transparent 25%),
-			linear-gradient(45deg, transparent 75%, #{$checkered-color} 75%),
-			linear-gradient(45deg, transparent 75%, #{$checkered-color} 75%),
-			linear-gradient(45deg, #{$checkered-color} 25%, #fff 25%);
-		background-size: 2 * $checkered-size 2 * $checkered-size;
-		background-position: 0 0, 0 0, -#{$checkered-size} -#{$checkered-size}, $checkered-size $checkered-size;
-	}
```

Comment:
This is a regression. We want to be able to see transparent backgrounds.

Reply:
Thanks, I see how this was needed.

However, the problem is that in order for the image to scale up (still keeping aspect ratio, see my other comment), the image has `100%` width and height so the white/checkered background will fill the entire screen. I'm not sure how to fix it since I don't think CSS can know the 'box' of the image.